### DISCUSSION TITLE
mdns broadcasting all interfaces while I want to bind only specified one

### DISCUSSION THREAD

**Discussion by eldereal** [libp2p/rust-libp2p#5599] [General]
I want my service bind only some interfaces. I can achieve it by:
```
swarm.listen_on(format!("/ip4/192.168.1.100/tcp/12345").parse()?)?;
```
By this, the service only listens on 192.168.1.100. But when I enable mdns service, It still broadcasting all interface addresses.

How to restrict mdns only specified addresses in such case? I think this should be a common issue, maybe I missed something. But I searched the document and sources and find no clues.

**Comment by dariusc93** [comment]
Hey @eldereal. I may be wrong, but reviewing the code it looks like this would likely require changing `InterfaceState` so it would use an explicit addresses instead of an unspecified address. 

https://github.com/libp2p/rust-libp2p/blob/a2a281609a0a64b211f7917aa856924983b63200/protocols/mdns/src/behaviour/iface.rs#L121-L154

Any reason why you would like to use a specific address?

**Reply by eldereal** [reply]
Thank you @dariusc93  for your reply. My case is the server has two interfaces. One is public and one is private. I want peers only works in private network and do not expose data to public network. I will have a try to rewrite this code.

**Reply by drHuangMHT** [reply]
My idea would be exposing attack surface as little as possible. I'm also looking forward to this change!
